src/_premake_init.lua

```diff
@@ -169,6 +169,12 @@
 		}
 	}
 
+	api.register {
```

Comment:
As it seems visualstudio specific, put it in vstudio/_preload.lua instead.
Or maybe other generators "just" have to call `clang-tidy source.cpp $(CXXFLAGS)` after `cxx source.cpp $(CXXFLAGS)`...

Reply:
> As it seems visualstudio specific, put it in vstudio/_preload.lua instead.

This suggestion is easier, and something I have now added a commit for. Perhaps clang-tidy for other generators can be future work.